modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -887,9 +900,14 @@ else if (!FAILED.equals(workflow.getState())) {
 
   @Override
   public WorkflowDefinition getWorkflowDefinitionById(String id) throws NotFoundException {
-    WorkflowDefinition def = workflowDefinitionScanner.getWorkflowDefinition(id);
-    if (def == null)
-      throw new NotFoundException("Workflow definition '" + id + "' not found");
+    final WorkflowIdentifier workflowIdentifier = new WorkflowIdentifier(id, securityService.getOrganization().getId());
+    WorkflowDefinition def = workflowDefinitionScanner.getWorkflowDefinition(workflowIdentifier);
+    if (def == null) {
+      def = workflowDefinitionScanner.getWorkflowDefinition(workflowIdentifier.withoutOrganization());
```

Comment:
Can't getWorkflowDefinition() simply fall back to the global workflow definition of there isn't one for this specific organization instead of us having to ask twice?

Reply:
It's a valid point. However, no matter what you do, you're going to have code that checks twice. Currently, this code is in `WorkflowServiceImpl`, making `WorkflowDefinitionScanner` dumb. We could make `WorkflowDefinitionScanner` smarter and put the check in there, but then we'd lose the ability to get the definition without the fallback mechanism.
I vote for keeping it the way it is.